DEV-4/DEV-4/Discipline.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DEV_4
+{
+    /// <summary>
+    /// This class contains all discipline materials.
+    /// </summary>
+    class Discipline : ObjectWithIdAndDescription, ICloneable
+    {
+        public List<Lecture> lectures;
+        public List<Seminar> seminars;
+        public List<LaboratoryWork> labs;
+
+        /// <summary>
+        /// The class constructor.
+        /// </summary>
+        /// <param name="description">Description</param>
+        /// <param name="theme">Theme</param>
+        public Discipline(string description)
+            :base(description)
+        {
+            lectures = new List<Lecture>();
+            seminars = new List<Seminar>();
+            labs = new List<LaboratoryWork>();
+        }
+
+        /// <summary>
+        /// The private class constructor for making deep copy of this object.
+        /// </summary>
+        /// <param name="description">Description</param>
+        /// <param name="lectures">List of lectures</param>
+        /// <param name="seminars">List of seminars</param>
+        /// <param name="labs">List of labs</param>
+        /// <param name="id">ID</param>
+        private Discipline(string description, List<Lecture> lectures, List<Seminar> seminars, List<LaboratoryWork> labs, string id)
+            :this(description)
+        {
+            Id = id;
+            foreach(var lecture in lectures)
+            {
+                this.lectures.Add((Lecture)lecture.Clone());
+            }
+
+            foreach(var seminar in seminars)
+            {
+                this.seminars.Add((Seminar)seminar.Clone());
+            }
+
+            foreach(var lab in labs)
+            {
+                this.labs.Add((LaboratoryWork)lab.Clone());
+            }
+        }
+
+        /// <summary>
+        /// This method adds lecture to List od lectures.
+        /// </summary>
+        /// <param name="lecture">Lecture object</param>
+        public void AddLecture(Lecture lecture)
+        {
+            lectures.Add(lecture);
+        }
+
+        /// <summary>
+        /// This method returns all lections with specified theme.
+        /// </summary>
+        /// <param name="theme">Theme</param>
+        /// <returns>lList of lectures</returns>
+        public List<Lecture> GetLecturesByTheme(string theme)
+        {
+            var lectures = new List<Lecture>();
+            foreach(var lecture in this.lectures)
```

Comment:
please add blank lines between different blocks of code

var lectures;

foreach ()
{

}

return